### DISCUSSION TITLE
Is there any way to set the api key dynamically rather than through environment variables?

### DISCUSSION THREAD

**Discussion by fernhub** [llm-tools/embedJs#109] [Q&A]
Trying to use this to create a demo app on switching between conversations with different LLMs and want to allow users to input their own API Keys rather than having my own set through env variables

**Comment by adhityan** [comment]
Sorry for the late response. Yes, it's possible. Most model constructors allow you to pass the API key programmatically. Now that the major refactor to a monorepo setup is done - focus will be on quality of life features including ensuring all models can be configured programmatically.
